tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/ScalaJSClassEmitter.scala

```diff
@@ -250,36 +301,52 @@ private[scalajs] final class ScalaJSClassEmitter(
   }
 
   /** Generates the JS constructor for a class, ES6 style. */
-  private def genES6Constructor(tree: LinkedClass): js.Tree = {
+  private def genES6Constructor(tree: LinkedClass,
+      cache: InvalidatableCache): js.Tree = {
     implicit val pos = tree.pos
 
     if (tree.kind.isJSClass) {
-      val js.Function(params, body) = genConstructorFunForJSClass(tree)
+      val js.Function(params, body) = genConstructorFunForJSClass(tree, cache)
       js.MethodDef(static = false, js.Ident("constructor"), params, body)
     } else {
+      val className = tree.name.name
       val fieldDefs = genFieldDefs(tree)
-      if (fieldDefs.isEmpty && outputMode == OutputMode.ECMAScript6) {
+
+      val (otherStatements, args) = if (usesJSConstructorOpt(className)) {
+        val List(ctor) = tree.memberMethods.map(_.tree).filter(
+            x => isConstructorName(x.name.name))
+        val js.Function(args, ctorBody) = desugarToConstructor(this,
+            className, cache, ctor.args, ctor.body, js.Skip())
+        (ctorBody, args)
+      } else {
+        (js.Skip(), Nil)
+      }
+
+      if (fieldDefs.isEmpty && outputMode == OutputMode.ECMAScript6 &&
+          otherStatements.isInstanceOf[js.Skip]) {
```

Comment:
You can remove `&& outputMode == OutputMode.ECMAScript`. It's always true, since strong mode has disappeared.